fabtests/common/shared.c

```diff
@@ -435,7 +435,8 @@ static int ft_alloc_msgs(void)
 		ft_set_tx_rx_sizes(&tx_size, &rx_size);
 		tx_mr_size = 0;
 		rx_mr_size = 0;
-		buf_size = MAX(tx_size, FT_MAX_CTRL_MSG) + MAX(rx_size, FT_MAX_CTRL_MSG);
+		buf_size = (MAX(tx_size, FT_MAX_CTRL_MSG) + MAX(rx_size, FT_MAX_CTRL_MSG))
+					* opts.window_size;
```

Comment:
To avoid sharing buffers, you probably want something like:
```
buf_size = MAX(tx_size,.. ) * opts.window_size + MAX(rx_size,...) * opts.window_size;
```